src/lib/datamath/index.ts

```diff
@@ -28,19 +29,35 @@ export default class datamath {
         return [minimum, maximum]
     }
 
+    static singlePercent(
+        data: number,
+        [minv, maxv]: [number, number]
+    ) {
+        const scalesize = this.rangeSize([minv, maxv])
+
+        const offset = data - minv
+        return offset / scalesize
+    }
```

Comment:
We don't need to duplicate functions
use
`const [result] = datamath.percent([value], range)`